app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/catalog/ui/view/FirstPageContent.kt

```diff
@@ -0,0 +1,249 @@
+                .padding(
+                    start = dimensionResource(id = R.dimen.catalog_horizontal_margin),
+                    end = dimensionResource(id = R.dimen.catalog_horizontal_margin),
+                    bottom = dimensionResource(id = R.dimen.catalog_margin_small)
+                )
+                .layoutId(FIRST_PAGE_BOTTOM_PAGE_NUMBER_ID),
+            text = stringResource(
+                id = R.string.catalog_page_no,
+                CatalogPage.Page1.ordinal + 1,
+                catalogList.sizeOrZero()
+            )
+        )
+    }
+}
+
+@Composable
+private fun getMainConstraintSet() = ConstraintSet {
+    val contentRef = createRefFor(TABLE_OF_CONTENTS_ID)
+    val bottomPageNumberRef = createRefFor(FIRST_PAGE_BOTTOM_PAGE_NUMBER_ID)
+
+    constrain(contentRef) {
+        linkTo(start = parent.start, end = parent.end)
+        linkTo(top = parent.top, bottom = bottomPageNumberRef.top)
+    }
+
+    constrain(bottomPageNumberRef) {
+        start.linkTo(parent.start)
+        bottom.linkTo(parent.bottom)
+    }
+}
+
+@Composable
+private fun getConstraintSetForTableOfContents() = ConstraintSet {
+    val firstTextRef = createRefFor(FIRST_PAGE_FIRST_TEXT_ID)
+    val secondTextRef = createRefFor(FIRST_PAGE_SECOND_TEXT_ID)
+    val thirdTextRef = createRefFor(FIRST_PAGE_THIRD_TEXT_ID)
+    val fourthTextRef = createRefFor(FIRST_PAGE_FOURTH_TEXT_ID)
+    val fifthTextRef = createRefFor(FIRST_PAGE_FIFTH_TEXT_ID)
+
+    constrain(firstTextRef) {
+        start.linkTo(parent.start)
+        top.linkTo(parent.top)
+    }
+
+    constrain(secondTextRef) {
+        end.linkTo(parent.end)
+        start.linkTo(parent.start)
+        top.linkTo(firstTextRef.bottom)
+    }
+
+    constrain(thirdTextRef) {
+        end.linkTo(parent.end)
+        start.linkTo(parent.start)
+        top.linkTo(secondTextRef.bottom)
+    }
+
+    constrain(fourthTextRef) {
+        end.linkTo(parent.end)
+        start.linkTo(parent.start)
+        top.linkTo(thirdTextRef.bottom)
+    }
+
+    constrain(fifthTextRef) {
+        end.linkTo(parent.end)
+        start.linkTo(parent.start)
+        top.linkTo(fourthTextRef.bottom)
+    }
+}
+
+@Composable
+fun TableOfContents(modifier: Modifier = Modifier, catalogItem: CatalogItem) {
+    val constraintSet = getConstraintSetForTableOfContents()
+
+    ConstraintLayout(
+        constraintSet = constraintSet,
+        modifier = modifier.fillMaxSize()
+    ) {
+        Text(
+            modifier = Modifier
+                .padding(
+                    start = dimensionResource(id = R.dimen.catalog_horizontal_margin),
+                    top = dimensionResource(id = R.dimen.catalog_margin_very_large),
+                    end = dimensionResource(id = R.dimen.catalog_horizontal_margin)
+                )
+                .layoutId(FIRST_PAGE_FIRST_TEXT_ID),
+            text = catalogItem.primaryDescription,
+            style = TextStyle(
+                color = MaterialTheme.colors.onSurface,
```

Comment:
May I please suggest moving all the `TextStyle` to the `Theme` file, since there are quite a few to improve the future maintenance?